tests/utils/gossamer_utils.go

```diff
@@ -357,18 +365,31 @@ func InitializeAndStartNodesWebsocket(t *testing.T, num int, genesis, config str
 			if i < len(KeyList) {
 				name = KeyList[i]
 			}
-			node, err := RunGossamer(t, i, TestDir(t, name), genesis, config, true, false)
-			if err != nil {
-				Logger.Errorf("failed to run Gossamer for node index %d", i)
+			node, runErr := RunGossamer(t, i, TestDir(t, name), genesis, config, true, false)
+			if runErr != nil {
+				errMutex.Lock()
+				if err == nil {
+					err = fmt.Errorf("failed to run Gossamer for node index %d: %w", i, runErr)
+				}
+				errMutex.Unlock()
+				return
 			}
 
+			nodesMutex.Lock()
 			nodes = append(nodes, node)
+			nodesMutex.Unlock()
+
 			wg.Done()
```

Comment:
move this `wg.Done()` to the goroutine start as `defer wg.Done()`

Reply:
Awesome, thanks!